### PR TITLE
Package file to build stand alon desktop application.

### PR DESCRIPTION
**Pull request by raul-ortega**
As you should know, Chrome Web Store application will support no more Windows, OS X and Linux systems. This file is needed to build the stand alone version with nw.js

Instructions to build with nw.js are here:

[https://github.com/nwjs/nw.js/wiki/How-to-package-and-distribute-your-apps](https://github.com/nwjs/nw.js/wiki/How-to-package-and-distribute-your-apps)

If you would like to test the result, I cloned the respository and bild the win32 version:

[https://github.com/raul-ortega/openLRSng-configurator/releases/tag/3.8.8](https://github.com/raul-ortega/openLRSng-configurator/releases/tag/3.8.8)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "width": 1024,
+    "height": 800
+  },
+  "repository": {
+    "type": "git",
+    "url": "github.com/openLRSng/openLRSng-configurator"
+  },
+  "author": "openLRSng",
+  "license": "GPL-3.0",
+  "dependencies": {
```
Comment (mikeller) [inline package.json]: That's an awful lot of runtime dependencies. Are you sure that all of these (including `gulp`, `nw-builder`) are required at runtime?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "openLRSng-configurator",
+  "description": "openLRSng Configurator",
+  "version": "3.8.8",
+  "main": "main.html",
+  "default_locale": "en",
+  "scripts": {
+    "start": "node_modules/gulp/bin/gulp.js build && node_modules/nw/bin/nw ."
```
Comment (mikeller) [inline package.json]: This fails for me:

```
user@pc:~/git/openLRSng-configurator$ npm start

> openLRSng-configurator@3.8.8 start /home/user/git/openLRSng-configurator
> node_modules/gulp/bin/gulp.js build && node_modules/nw/bin/nw .

[13:34:27] No gulpfile found
```

Can you please fix?

Comment (raul-ortega) [reply package.json]: I first included the package.json file in the openlrsng repository files. Then y created a zip file manually selecting all files (and folders), and rename it to openLRSng-configurator.nw.

After dat I went to the nw.js respository and downloaded the were I had decompressed the nwjs zip file for win32: [https://github.com/nwjs/nw.js#downloads](https://github.com/nwjs/nw.js#downloads), and unziped it.

Then I moved the openLRSng-configurator.nw file into the nwsjs folder and executed this command from cmd:

copy /b nw.exe+openLRSng-configurator.nw openLRSng-configurator.exe

After that I removed files nw.exe and openLRSng-configurator.nw, then I zipped everything toguether and uploaded it to the releases folder of the repository.

Thas all, and is exactly the same that betaflight or iNav standalone apps:

betaflight-vs-openlrsng-standalone

File: package.json (hunk 1 above)
Comment (raul-ortega) [reply package.json]: Please, see my previous answer to your first reply.

Comment (raul-ortega) [reply package.json]: Take a look to iNav and betaflight package.json files:

[https://github.com/iNavFlight/inav-configurator/blob/master/package.json](https://github.com/iNavFlight/inav-configurator/blob/master/package.json)

[https://github.com/betaflight/betaflight-configurator/blob/master/package.json](https://github.com/betaflight/betaflight-configurator/blob/master/package.json)

File: package.json (hunk 2 above)
Comment (mikeller) [reply package.json]: I can't quite see how this description relates to the fact that the `start` script in your script seems to be broken. If it's meant to work it should be fixed, if it's not meant to work it should not be in there in the first place.

Also I think that having to do these steps manually will lead to poor repeatability for building / testing, and should be avoided.

Have a look at https://github.com/betaflight/betaflight-configurator for an example of how the build can be completely automated `npm start` will actually build, and then start the configurator in Betaflight.

File: package.json (hunk 1 above)
Comment (mikeller) [reply package.json]: This still does not explain why your `package.json` is declaring so many runtime dependencies, with a lot of them looking like they should be dev dependencies.

Comment (raul-ortega) [reply package.json]: Oh! I apologyze. You can remove the dependencies and script sections, it is only needed to build from development version.

Finally the package.json would be like this:

{
  "name": "openLRSng-configurator",
  "description": "openLRSng Configurator",
  "version": "3.8.8",
  "main": "main.html",
  "default_locale": "en",
  "window": {
    "title": "openLRSng Configurator",
    "icon": "images/icon_128.png",
    "toolbar": true,
    "width": 1024,
    "height": 800
  },
  "repository": {
    "type": "git",
    "url": "github.com/openLRSng/openLRSng-configurator"
  },
  "author": "openLRSng",
  "license": "GPL-3.0"
}

Comment (mikeller) [reply package.json]: In this case you are not installing any dependencies. This will certainly not work, since things like `nw-builder` are required to **build** the standalone app - they just are not required to **run** the standalone app.

I suggest we close this, and wait until me (or somebody else) ports the build system that is used for the Betaflight configurator to the openLRSng configurator. It is on top of my priority list, and there are a couple of holidays coming up...

File: package.json (hunk 2 above)
Comment (raul-ortega) [reply package.json]: I agree with you about the need to automate the building process for developing. If I have more time I'll help you doing it. By now I only wanted to make a litle contribution to help end users to run the configurator without the need of Google Chrome. I thik you could build it manually for the last version and publish it, so future users could download and run it.

Comment (mikeller) [reply package.json]: Thank you for trying to help.

At this point I don't see much value in adding an essentially empty `package.json` (once you've removed the things that are broken or unneeded), and then expecting users to manually install all of the parts of the toolchain and do the build.
If they want to do the build manually, they can just as well execute a `npm init` as the first step, in order to create an empty `package.json`.

Comment (raul-ortega) [reply package.json]: I don't want end users to build it. What I mean is to suggesst to you to build it manually and publish it under releases section for future users, until the developers automate it.

Comment (mikeller) [reply package.json]: You are suggesting we manually install and then manually build / pack the app on 3 platforms that should be supported (linux, macos, windows). And you hope that we will get it right and manage to produce 3 identical builds. And then we do this again all over, if a bug is reported and fixed?

This will require a lot of time (and access to 3 different build machines). Time that could otherwise be spent on implementing an automated build.

Another consideration is that the app in it's current state is not ready to work as a standalone app: It relies on one firmware version being packaged inside the app, and the app only being able to install this one version, or a custom built, not verified binary. This was a good approach while the app was delivered through the Chrome web store, since it essentially leveraged the web store to push the firmware onto the users' workstations. For the standalone app this won't work - we can't expect users to download and install a new release of the > 80 MB standalone application just to get a new firmware release. Have a look at Betaflight configurator to see how it is capable of getting the latest firmware release over the internet, in a user friendly way. This is what we should be going for in the standalone app.

So to sum it up, I don't see why we should be offering users an incomplete and unreliable standalone app - they will be better off if they continue to use the Chrome web store app until the standalone app (including the build process) is finalised and ready for release.

Comment (raul-ortega) [reply package.json]: And when do you spect to be released???

Comment (mikeller) [reply package.json]: When it's ready for release - within the next month or so.

Comment (raul-ortega) [reply package.json]: Ok I see reasonable to wait for then.
